Amethyst/Managers/WindowManager.swift

```diff
@@ -21,13 +23,111 @@ enum WindowChange {
     case unknown
 }
 
-private struct ObserveApplicationNotifications {
+enum MouseState {
+    case pointing
+    case clicking
+    case dragging
+    case moving(window: SIWindow)
+    case resizing(screen: NSScreen, ratio: CGFloat)
+    case doneDragging(atTime: NSDate)
+}
+
+class MouseStateKeeper {
+    public var state: MouseState
+    private var _windowManager: WindowManager!
+    private var monitor: Any?
+
+    init() {
+        state = .pointing
+    }
+
+    var windowManager: WindowManager {
+        get {
+            return _windowManager
+        }
+        set {
+            self._windowManager = newValue
+            let mouseEventsToWatch: NSEvent.EventTypeMask = [.leftMouseDown, .leftMouseUp, .leftMouseDragged]
+
+            // remove any existing monitors, although we expect to never reassign windowManager.
+            if let oldMonitor = monitor {
+                NSEvent.removeMonitor(oldMonitor)
+                monitor = nil
+            }
+
+            monitor = NSEvent.addGlobalMonitorForEvents(matching: mouseEventsToWatch) { anEvent in
```

Comment:
I think this is also a retain loop. The block maintains a strong reference to `self` so it'll never deinit to remove the monitor. In practice this isn't an issue because there only needs to be one for the lifetime of the application, but things might get weird for testing purposes.